Library/Homebrew/rubocops/keg_only.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      class Keg < FormulaCop
+        def audit_formula(node, _class_node, _parent_class_node, _body_node)
+          return unless find_node_method_by_name(node, :keg_only?)
+
+          offending_node(node)
+
+          whitelist = %w[
+            Apple
+            macOS
+            OS
+            Homebrew
+            Xcode
+            GPG
+            GNOME
+            BSD
+            Firefox
+          ].freeze
+
+          reason = find_node_method_by_name(_body_node, keg_only_reason).to_s
```

Comment:
The reason actually looks like:
```ruby
keg_only "conflicts with other Glk libraries"
```

where the latter string is the `keg_only_reason`